pkg/controllers/node/controller.go

```diff
@@ -0,0 +1,92 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha3"
+	"go.uber.org/multierr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client) *Controller {
+	return &Controller{
+		kubeClient: kubeClient,
+	}
+}
+
+// Controller manages a set of properites on karpenter provisioned nodes, such as
+// taints, labels, finalizers.
+type Controller struct {
+	kubeClient client.Client
+	readiness  *Readiness
+	finalizer  *Finalizer
+}
+
+// Reconcile executes a reallocation control loop for the resource
+func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("Node"))
+	// 1. Retrieve node from reconcile request
+	stored := &v1.Node{}
+	if err := c.kubeClient.Get(ctx, req.NamespacedName, stored); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	if len(stored.Labels[v1alpha3.ProvisionerNameLabelKey]) == 0 {
+		return reconcile.Result{}, nil
+	}
+
+	// 2. Execute node reconcilers
+	node := stored.DeepCopy()
+	var errs error
+	for _, reconciler := range []interface {
+		Reconcile(*v1.Node) error
+	}{
+		c.readiness,
+		c.finalizer,
+	} {
+		errs = multierr.Append(errs, reconciler.Reconcile(node))
```

Comment:
Is there a point to keeping these errs if we're not going to do anything with them?

Reply:
Yikes!

Reply:
Fixed